framework/resource_router.go

```diff
@@ -37,7 +51,11 @@ func (r *ResourceRouter) ResourceSet(obj interface{}) (*ResourceSet, error) {
 	}
 
 	if len(found) == 0 {
-		return nil, microerror.Maskf(executionFailedError, "handling resource set not found")
+		k, err := cache.MetaNamespaceKeyFunc(obj)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		r.logger.Log("level", "debug", "message", "no resource set for reconciled object", "key", k)
```

Comment:
Cool you extracted some info about the object here. I don't think returning error is necessary in this case, but it is ok.

I'm not a fan of entrancing new random key-values for the logger though. 

This is how we do it in resource_set.go

```
	accessor, err := meta.Accessor(obj)
	if err != nil {
		r.logger.LogCtx(ctx, "function", "InitCtx", "level", "warning", "message", "cannot create accessor for object", "object", fmt.Sprintf("%#v", obj), "stack", fmt.Sprintf("%#v", err))
	} else {
		meta, ok := loggermeta.FromContext(ctx)
		if !ok {
			meta = loggermeta.New()
		}
		meta.KeyVals["object"] = accessor.GetSelfLink()

		ctx = loggermeta.NewContext(ctx, meta)
	}
```

I think it would be good to reuse `ojbject` key and set it to the same value (self link)

Reply:
Good shout. Done. Please check again.